File: impls/java/Cat.java
import java.io.IOException;

public class Cat {

    public static void main(String[] args) {
        while (true) {
            int n = readByte();
            if (n == /*EOF*/-1) {
                break;
            }
            System.out.write(n);
        }

        System.out.flush();
    }

    private static int readByte() {
        try {
            return System.in.read();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

}

File: impls/erlang/cat.erl
-module(cat).
-export([main/0]).

main_iter(IoDevice) ->
    case file:read(IoDevice, 1) of
        eof -> nil;
        {error, Reason} -> io:fwrite("~s~n", [Reason]);
        {ok, [Byte]} ->
            io:fwrite("~c", [Byte]),
            main_iter(IoDevice)
    end.

main() ->
    io:setopts(standard_io, [{encoding, latin1}]),
    main_iter(standard_io).
